### ISSUE TITLE
Dotnet core?

### ISSUE THREAD

**Issue by interdrift** [YaccConstructor/QuickGraph#149]
Do you have any plans to port Quickgraph to be usable in dotnet core?

**Comment by gsvgit** [comment]
Yes, we have. But in mind only. First of all we should solve problems with "legacy" PEX-based tests (Issue #5), CodeContracts, and may be with PCL (Issue #6).  More over, seems that [F# is not fully supported](https://github.com/dotnet/core/blob/master/roadmap.md), migration is not actual just now because there are many algorithms implemented in F#. Yes, [we can use F#] (https://docs.microsoft.com/en-us/dotnet/articles/fsharp/getting-started-netcore) but it is a challenge.

I think that port QuickGraph to .Net Core is good idea, but we will work on it a bit later.

**Comment by CumpsD** [comment]
Any progress on this? Now that Core is RTM and already iterating over new versions

**Comment by DashNY** [comment]
+1 Would love to use in a .NET Standard library. Thank you.

**Comment by legnars** [comment]
@gsvgit hello! How's the progress on this? Anyway I can actively support the changes to allow .NET Core?

**Comment by gsvgit** [comment]
@legnars hello! Seems, that we still can get problems with CodeContracts (seems that it is not ported to .NET Core: https://github.com/Microsoft/CodeContracts/issues/231) Could you do some "research" to create a detailed list of problems?

If we still have serious problems, we should discuss it. Maybe we should find replacement for PEX and CodeContracts or just drop them out (but the last case is not preferable).

**Comment by legnars** [comment]
@gsvgit based on this commentary
https://github.com/Microsoft/CodeContracts/issues/493#issuecomment-353752918
I think it's unlikely that CodeContracts will be supported in .NET Core/VS 2017 any time soon.
(So not only it has problems compiling in core, it's also not officially supported in VS 2017)

**Comment by irperez** [comment]
I agree with @legnars.  Code Contracts is a legacy framework.  Microsoft has moved on from that.

**Comment by gsvgit** [comment]
@legnars @irperez Well. Let just remove all CodeContracts related stuff. @legnars can you do it?

Next. What about PEX? It is a legacy too. Is it possible to migrate test to IntellyTest? As a first step, we can just save PEX-generated tests.

**Comment by legnars** [comment]
@gsvgit Yes, I can!
How should we validate the current contract conditions though?

I have not worked with PEX before! But if we can run them in .NET Framework it would be validating the same code as in .NET Core, right? So it can wait while we migrate the tests.

**Comment by gsvgit** [comment]
@legnars Can you look at https://github.com/deniszykov/code-contracts-remover ? It may help us to convert CodeContracts' conditions to "if (cond) then ... else throw ...". I think that it is one of the more appropriate ways of conditions conversion.

On PEX. Yes, it can wait. PEX generates tests for compiled binaries. As far as .NET Core binary format (instructions set) has differences from .NET Framework, it may be impossible to use PEX for .NET Core project. So, my proposal is:
1) Run PEX on .NET Framework project to generate tests
2) Save these tests as standard (N|MS|Smth_else)Unit tests
3) Use saved tests in .NET Core project.

**Comment by legnars** [comment]
@gsvgit after removing CodeContracts from the project (and making the proper changes), 11 tests started failing:

- ChromaticPolinomialAlmostFullGraph
- ChromaticPolinomialEmptyGraph
- GraphPartitioningTest1
- GraphPartitioningTest2
- GraphPartitioningTest3
- IsEulerianManyComponentsFalse
- IsEulerianManyComponentsTrue
- IsEulerianOneComponentFalse
- IsEulerianOneComponentTrue
- IsEulerianTwoVerticesTwoEdges
- YenZeroCaseTest

They all break on this new line in the UndirectedEdge class:
```c#
if (Comparer<TVertex>.Default.Compare(source, target) > 0) {  throw new System.ArgumentException("source", "Comparer<TVertex>.Default.Compare(source, target) <= 0"); }
```

Except for YenZeroCaseTest, which breaks on this new line in the DijkstraShortestPathAlgorithm class:
```c#
if (!(this.VisitedGraph.ContainsVertex(rootVertex))) {  throw new System.ArgumentException("rootVertex", "this.VisitedGraph.ContainsVertex(rootVertex)"); }
```

Could the new if-then logic be revealing past broken tests?

**Comment by abk-x** [comment]
Not to step on any toes,

I am using YC.QuickGraph via nuget in a netstandard2.0 project, and I am not seeing any issues using it (with exceptions of nuget warnings for DotNet.Contracts). It would be interesting to know if the tests are successful however.

@legnars are you running tests via dotnet core? or via .NET? If the former, if you could point me to the repo I am keen to have a look at those failing tests.

**Comment by legnars** [comment]
@abk-x I’m still running them via .NET Framework, I just removed all references to Code Contracts so far.


________________________________

**Comment by 1354092549** [comment]
Is there any progress?

**Comment by KeRNeLith** [comment]
Hello,

I forked this QuickGraph repository ([here](https://github.com/KeRNeLith/QuikGraph)) and put a lot of effort to make it fully .NET Core compliant. I extended support of the library and obviously created targets for .NET Core.

I added a huge set of unit tests that are all green.

I get rid of PEX for unit testing.
I get rid of CodeContracts that apparently are not well compiled in the release on NuGet. I tested and was able to do things that should normally break. So I replaced them by standard assertions and throws.

Side note: this massive refactor was only done for QuickGraph core project for now.

**Comment by vidyakdil** [comment]
when can we get Quickgraph compatible for .net core 3.1?
